torchtext/vocab.py

```diff
@@ -106,7 +105,7 @@ def _default_unk_index(self):
         return self.unk_index
 
     def __getitem__(self, token):
-        return self.stoi.get(token, self.stoi.get(Vocab.UNK))
+        return self.stoi.get(token, self.unk_index)
```

Comment:
Say ```specials=['<pad>']``` and ```unk_token='<unk>' ```, self.unk_index = None (initiated).

Then the __getitem__ func will return `None` for unknown tokens. Not sure if this is correct.

Reply:
You are correct, it should raise an error. Fixing.